daemon/network.go

```diff
@@ -1043,33 +1043,31 @@ func buildEndpointInfo(networkSettings *internalnetwork.Settings, n *libnetwork.
 		return nil
 	}
 
-	if _, ok := networkSettings.Networks[n.Name()]; !ok {
-		networkSettings.Networks[n.Name()] = &internalnetwork.EndpointSettings{
+	nwName := n.Name()
+	if _, ok := networkSettings.Networks[nwName]; !ok {
+		networkSettings.Networks[nwName] = &internalnetwork.EndpointSettings{
 			EndpointSettings: &network.EndpointSettings{},
 		}
 	}
-	networkSettings.Networks[n.Name()].NetworkID = n.ID()
-	networkSettings.Networks[n.Name()].EndpointID = ep.ID()
+	networkSettings.Networks[nwName].NetworkID = n.ID()
+	networkSettings.Networks[nwName].EndpointID = ep.ID()
 
-	iface := epInfo.Iface()
-	if iface == nil {
-		return nil
-	}
-
-	if iface.MacAddress() != nil {
-		networkSettings.Networks[n.Name()].MacAddress = iface.MacAddress().String()
-	}
+	if iface := epInfo.Iface(); iface != nil {
```

Comment:
nit: I don't like this change, it adds indentation and is different than the logic just above on line 1041

Reply:
Ah, yes, I agree; we can return early here. I can update tomorrow.